ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -84,7 +84,17 @@ public StructuredDataSource getSource(final String sourceName) {
     return dataSources.values()
         .stream()
         .filter(p -> p.source.getKsqlTopic().getName() != null
-            && p.source.getKsqlTopic().getName().equals(ksqlTopicName))
+            && p.source.getTopicName().equals(ksqlTopicName))
+        .map(sourceInfo -> sourceInfo.source)
+        .findFirst();
+  }
+
+  @Override
+  public Optional<StructuredDataSource> getSourceForKafkaTopic(final String kafkaTopicName) {
+    return dataSources.values()
+        .stream()
+        .filter(p -> p.source.getKsqlTopic().getName() != null
```

Comment:
what's this null check for? that's going to check the underlying ksql topic name (not the kafka topic name).

Reply:
Fixed, thanks for catching that!